libraries/c_sdk/standard/ble/test/iot_test_ble_end_to_end.c

```diff
@@ -107,13 +250,29 @@ TEST_GROUP_RUNNER( Full_BLE_END_TO_END_MQTT )
     RUN_TEST_CASE( Full_BLE_END_TO_END_MQTT, SubscribeCompleteReentrancy );
     RUN_TEST_CASE( Full_BLE_END_TO_END_MQTT, IncomingPublishReentrancy );
 
+
     /* Revert to sockets network interface after this test is finished. */
     IotTestNetwork_SelectNetworkType( DEFAULT_NETWORK );
 }
 
 
 /*-----------------------------------------------------------*/
 
+TEST( Full_BLE_END_TO_END_CONNECTIVITY, InitializeBLE )
+{
+    _BLEInit();
+}
+
+TEST( Full_BLE_END_TO_END_CONNECTIVITY, EnableBLE )
+{
+    _BLEEnable();
+}
+
+TEST( Full_BLE_END_TO_END_CONNECTIVITY, SetDeviceName )
+{
+    TEST_ASSERT_EQUAL( eBTStatusSuccess,
+                       IotBle_SetDeviceName( IOT_BLE_DEVICE_COMPLETE_LOCAL_NAME, strlen( IOT_BLE_DEVICE_COMPLETE_LOCAL_NAME ) ) );
```

Comment:
This test just checks whether or not `IotBle_SetDeviceName()` returns `eBTStatusSuccess`.  How do we know it actually did anything?

Reply:
This a system level test. The BLE peer will scan and connect to devices using the name set via this API and so the next test BLE_Enable() succeeds only if the device name is set correctly in the advertisement.